src/signTxOutput.c

```diff
@@ -532,18 +550,86 @@ static void signTxOutput_handleTokenAPDU(uint8_t* wireDataBuffer, size_t wireDat
 		txHashBuilder_addOutput_token(
 		        &BODY_CTX->txHashBuilder,
 		        subctx->stateData.token.assetNameBytes, subctx->stateData.token.assetNameSize,
-		        subctx->stateData.token.amount
+		        subctx->stateData.token.amount,
+		        subctx->includeDataHash
 		);
 		TRACE();
 	}
 
 	signTxOutput_handleToken_ui_runStep();
 }
 
+// ========================== SCRIPT DATA HASH =========================
```

Comment:
I find using "script data hash" here confusing. Officially, the thing being hashed is called "datum" (see https://iohk.io/en/research/library/papers/the-extended-utxo-model/), so I propose to call it "datum hash" everywhere, including in the UI (close, but not precisely equal, to "data_hash" used in CDDL), e.g. we use `HANDLE_DATUM_HASH_STEP_DISPLAY`. That way, no confusion with other script data hashes arise.

Reply:
We can use datum, but it's called script_data_hash in the CDDL. Which is _more_ official?

Reply:
Only "data_hash" in CDDL:
```
transaction_output =
  [ address
  , amount : value
  , ? data_hash : $hash32 ; New
  ]

```
Anyway, https://docs.cardano.org/plutus/datums-redeemers also uses "datum"; CDDL is known to be slightly inconsistent with other docs from time to time, and has been modified at least once on our request because it is easy to change the CDDL (unlike the other docs).